base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -1888,97 +1892,159 @@ where
+                            .find(|x| PublicKey::from_secret_key(&x.private_key) == **spending_key_public)
+                        {
+                            None => None,
+                            Some(x) => {
+                                match PrivateKey::from_bytes(
+                                    CommsPublicKey::shared_secret(&x.private_key, &output.sender_offset_public_key)
+                                        .as_bytes(),
+                                ) {
+                                    Ok(spend_key) => Some((output.clone(), x.private_key.clone(), spend_key)),
+                                    Err(_) => None,
+                                }
                             },
                         }
-                        trace!(
-                            target: LOG_TARGET,
-                            "One-sided payment Output {} with value {} recovered",
-                            output_hex,
-                            committed_value,
-                        );
+                    },
+
+                    // ----------------------------------------------------------------------------
+                    // one-sided stealth address
+                    // NOTE: Extracting the nonce R and a spending (public aka scan_key) key from the script
+                    // NOTE: [RFC 203 on Stealth Addresses](https://rfc.tari.com/RFC-0203_StealthAddresses.html)
+                    [Opcode::PushPubKey(nonce), Opcode::Drop, Opcode::PushPubKey(provided_spending_public)] => {
+                        let secret_key = self.node_identity.secret_key().clone();
+                        let public_key = self.node_identity.public_key();
+
+                        // computing shared secret
+                        let c = RistrettoSecretKey::from_bytes(
+                            DomainSeparatedHasher::<Blake256, GenericHashDomain>::new("stealth address")
+                                .chain(PublicKey::shared_secret(&secret_key, nonce.as_ref()).as_bytes())
+                                .finalize()
+                                .as_ref(),
+                        )
+                        .unwrap();
+
+                        // calculating spending (public) key
+                        let ks = PublicKey::from_secret_key(&c) + public_key;
+
+                        if &ks != provided_spending_public.as_ref() {
+                            return None;
+                        }
+
+                        match PrivateKey::from_bytes(
+                            CommsPublicKey::shared_secret(&secret_key, &output.sender_offset_public_key).as_bytes(),
+                        ) {
+                            Err(_) => None,
+                            Ok(spend_key) => Some((output.clone(), secret_key.clone() + c, spend_key)),
+                        }
+                    },
+
+                    _ => None,
+                }
+            })
+            .collect_vec();
```

Comment:
Use this and remove itertools as dependency

```suggestion
            .collect::<Vec<_>>();
```

Reply:
it's a leftover from what I didn't use after all, though it's still nice to have around, and besides this service is already using it in dibbler. do you still want it out?